src/app/_core/services/weather.service.ts

```diff
@@ -135,9 +134,31 @@ export class WeatherService {
       )
   }
 
+  getWeatherCoordApi(lat: string, lon:string, unit: WeatherUnit, lang: string):Observable<CurrentWeather>{
+    let params = new HttpParams()
+      .set('lat', lat)
+      .set('lon', lon)
+      .set('unit', unit)
+      .set('lang', lang)
+      .set('appid', this.apiKey)
+    return this.http.get<WeatherApiResponse>(API_WEATHER_URL, {params})
+      .pipe(
+        map(response => constructWeatherFromApiData(response, lang, unit))
```

Comment:
Can you refactor this method the same way?

- [ ] Remove CurrentWeather
- [ ] Remove WeatherApiResponse
- [ ] Remove construc.... (use contructor)